File: maximum-number-of-pairs-in-array/maximum-number-of-pairs-in-array.py
class Solution:
    def numberOfPairs(self, nums: List[int]) -> List[int]:
        res = []
        pairs=0
        num = list(set(nums))
        for i in range(len(num)):
            cnt = nums.count(num[i])
            if cnt%2!=0:
                res.append(num[i])
            pairs += cnt//2
        return [pairs,len(res)]
        

File: maximum-number-of-pairs-in-array/maximum-number-of-pairs-in-array.cpp
class Solution {
public:
    vector<int> numberOfPairs(vector<int>& nums) {
        unordered_map<int,int>res;
        for(int i = 0 ; i < nums.size() ; i++){
            res[nums[i]]++;
        }
        int cnt = 0 , cnt1 = 0 ;
        for(auto it: res){
            int a  = it.second;
            cnt1 += a%2;
            cnt += a/2;
        }
        return {cnt,cnt1};
        
    }
};